ui/ui-guidelines.md

```diff
@@ -1,17 +1,60 @@
 # UI Guidelines
 
-## Background
-
-This lays out the current state of how the frontend code is architected, the best
-practices to follow when adding new code to the frontend and the considerations
+This document lays out the current state of how the frontend code is architected, the best
+practices to follow when adding new code to the frontend, and the considerations
 for the future growth of the frontend. This is meant to be a living document that
-will evolve as the UI grows. There are lots of exceptions to every rule laid out
+will evolve as the UI grows.
+
+There are lots of exceptions to every rule laid out
 here as UI code can vary significantly from feature to feature.
 
-## Elements of the UI
+> ⚠️ You may find inconsistencies with some of the best practices in older code that was written before these guidelines were updated. New code should follow these guidelines. We are slowly working on bringing older code in line with the best practices.
```

Comment:
I've put a few of these little notes throughout because it can be really confusing to see best practices with a codebase that doesn't align with them. I'm hoping this communicates that the best practices are real and you should use them in new code.

Reply:
I've run across the following in some of the older code that we need to do some work to bring in line with the guidelines:
- Lack of unit test coverage for some of our utility code.
- Lack of integration and/or e2e test coverage for some of the UI. 
- Forgetting to export the prop types (e.g. `MyComponentProps`) for some exported components. 
- We have pretty limited documentation at this time in both Storybook and jsdoc on types/functions/classes. 
- I think we still have one place (I think it's in the variables provider) where you can't opt-out of using the predefined query params.

For most of these, we're trying to slowly fix them via the "campsite rule" (leave things better than you found then) when we touch that code to fix bugs or add new functionality. We'll probably need to do some more targeted work to make it possible to opt out of use-query-params for embedded use cases.